### PR TITLE
diff: specify available terminal width by caller, subtract graph width

### PR DESCRIPTION
**Pull request by yuja**
#4158

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: cli/tests/test_log_command.rs [hunk 1]
```diff
@@ -1377,6 +1377,43 @@ fn test_log_word_wrap() {
...
+    0 files changed, 0 insertions(+), 0 deletions(-)
+    "###);
+
+    // Graph width should be subtracted
+    insta::assert_snapshot!(render(&["log", "--stat"], 30), @r###"
+    @  qpvuntsm test.user@example.com 2001-02-03 08:05:08 e292def1
+    │  (no description set)
+    │  file1 | 100 ++++++++++++
+    │  1 file changed, 100 insertions(+), 0 deletions(-)
+    ◆  zzzzzzzz root() 00000000
```
Comment (jonathantanmy) [inline cli/tests/test_log_command.rs:1412]: Can you add another test or expand this test to have multiple nesting levels?

**Review by jonathantanmy — COMMENTED** [review]
Thanks, overall this looks good except for one comment above.

File: cli/tests/test_log_command.rs (hunk 1 above)
Comment (yuja) [reply cli/tests/test_log_command.rs:1422]: Sure. Added one more branch.

**Review by jonathantanmy — APPROVED** [review]
Thanks! Looks good.

**Pull request merged**